src/Rule/Callback.php

```diff
@@ -12,24 +12,61 @@ class Callback extends Rule
 {
     private $callback;
 
-    public function __construct(callable $callback)
+    public function __construct($callback)
     {
+        if (!(is_callable($callback) || (is_array($callback) && count($callback) === 2))) {
```

Comment:
Why did you do it?
http://sandbox.onlinephpfunctions.com/code/13996f1b47b5103604d7f4263786eb240a31f996

Reply:
you're right, `is_callable` will be enough

Reply:
to use private method we can pass the closure like the following:
```php
class Form 
{
    public function validate()
    {
        $rule = new Callback(function ($value) {
            $this->checkValue($value);
        });
        // or shortly
        $rule = new Callback(fn ($value) => $this->checkValue($value));
    }

    private function checkValue(string $value)
    {
        // some check
    }
}
```